static/js/overlays.js

```diff
@@ -320,28 +320,40 @@ export function open_settings() {
     });
 }
 
-export function initialize() {
-    $("body").on("click", "div.overlay, div.overlay .exit", (e) => {
-        let $target = $(e.target);
+export function handle_body_click(e) {
+    let $target = $(e.target);
 
-        if (document.getSelection().type === "Range") {
-            return;
-        }
+    if (document.getSelection().type === "Range") {
+        return;
+    }
+
+    // if the target is not the div.overlay element, search up the node tree
+    // until it is found.
+    if ($target.is(".exit, .exit-sign, .overlay-content, .exit span")) {
+        $target = $target.closest("[data-overlay]");
+    } else if (!$target.is("div.overlay")) {
+        // not a valid click target then.
+        return;
+    }
 
-        // if the target is not the div.overlay element, search up the node tree
-        // until it is found.
-        if ($target.is(".exit, .exit-sign, .overlay-content, .exit span")) {
-            $target = $target.closest("[data-overlay]");
-        } else if (!$target.is("div.overlay")) {
-            // not a valid click target then.
-            return;
-        }
+    const target_name = $target.attr("data-overlay");
 
-        const target_name = $target.attr("data-overlay");
+    close_overlay(target_name);
 
-        close_overlay(target_name);
+    e.preventDefault();
+    e.stopPropagation();
+}
 
-        e.preventDefault();
-        e.stopPropagation();
+export function enable_body_click() {
+    $("body").on("click", "div.overlay, div.overlay .exit", (e) => {
+        handle_body_click(e);
     });
 }
+
+export function disable_body_click() {
+    $("body").off("click", "div.overlay, div.overlay .exit");
+}
```

Comment:
I'm a little reluctant to merge something that makes the general overlays click behavior dependant on the `panzoom` library correctly calling its `panzoomend` / `panzoomstart` handlers.  (E.g. I can imagine that bug having a bug where it double-calls the `end` function).

Would it be possible to implement this detail by instead adding a handler for just the lightbox container for clicks on `div.overlay` (it doesn't seem like the exit element would need it), and just checks whether a flag for "are we dragging" that is managed by the `panzoomstart` / `panzoomend` functions is true, and if so, does an `e.stopPropagation()` to prevent this general overlays logic from running?

Reply:
Definitely struggling with this one a bit. I've tried several iterations but the click event is always fired on the parent overlay div when triggered at the end of a drag, even if it's located on a child element where a normal event can be caught and thus not propagated. If I try to just override the event handling on `div.overlay` as suggested in your comment, the general logic from `overlay.js` will always run first because it is registered first.

Reply:
OK, I went with a different approach that I think works better. Instead of enabling/disabling the click handlers, the pan/zoom events now modify a `data-noclose` data attribute on the overlay. If the `data-noclose` attribute is present and true, the click event handler responsible for closing the overlay will instead leave it open.

Reply:
Hmm, what do you see as the benefit of storing this state in a data attribute over a JavaScript variable?  It seems to me that a JavaScript variable would be a bit easier to reason about, since the logic around this state machine would all be in one place.

Reply:
Right now `lightbox.js` handles the pan/zoom stuff but `overlays.js` contains the code that enables the behavior to close the modal when an empty area is clicked. That is the behavior we need to stop when the user is dragging.

This means the state in `lightbox.js` needs to override behavior in `overlays.js` and importing anything from `lightbox.js` into `overlays.js` would be a circular dependency. I'm not sure how we would do it without a global. Even then, it seems cleaner to use the data attribute as it's more generic. Conceivably there are other cases where you'd want to shut off the "click empty space to close" behavior and with this code you can simply add the data attribute in the markup.

Reply:
Yeah, OK, that's a good reason to do it this way.